compiler/rustc_builtin_macros/src/deriving/generic/mod.rs

```diff
@@ -1101,7 +1111,7 @@ impl<'a> MethodDef<'a> {
         let variants = &enum_def.variants;
 
         // Traits that unify fieldless variants always use the tag(s).
-        let uses_tags = self.unify_fieldless_variants;
+        let uses_tags = self.unify_fieldless_variants == UnifyFieldlessVariants::Tag;
```

Comment:
I think the name `unify_fieldless_variants` is actually clearer than `uses_tags`:
```
let unify_fieldless_variants = FieldlessVariantsStrategy::Unify;
```